code/server/routes/api.js

```diff
@@ -127,8 +115,13 @@ function getProfileDate(req) {
 		dateOfBirth: req.body.dateOfBirth};
 }
 
+// Initialize as either 'db' or 'src'
+function initializeDAOs(mode) {
```

Comment:
Yeah that works for me, my idea was that main.js should 'initialize' the API, but you're right that the DAOs should be independent of main.js.